src/Core/Context.idr

```diff
@@ -1435,6 +1435,11 @@ hide fc n
               | res => ambiguousName fc n (map fst res)
          put Ctxt (record { gamma $= hideName nsn } defs)
 
+-- `unhide` tries to use `recordWarning`, which is defined below
+-- so I have to forward-declare it here...
+export
+recordWarning : {auto c : Ref Ctxt Defs} -> Warning -> Core ()
```

Comment:
Can't you just move the whole definition rather than implicitly declaring a mutual block?

Reply:
Ok, moved `unhide` to the bottom (and `hide` too, to have them next to each other)